src/graphql/utils.ts

```diff
@@ -0,0 +1,29 @@
+import { ContentFields } from 'contentful-management/types';
+
+export function logUnrecognizedFields(
+  contentTypeFields: string[],
+  data: Record<string, unknown>
+): void {
+  const recognized = new Set(['sys', '__typename', 'contentfulMetadata', ...contentTypeFields]);
+
+  for (const field of Object.keys(data)) {
+    if (!recognized.has(field)) {
+      console.warn(`Unrecognized field '${field}'. Note that GraphQL aliases are not supported`);
+    }
+  }
+}
+
+export function isPrimitiveField(field: ContentFields): boolean {
+  const types = new Set(['Symbol', 'Text', 'Integer', 'Boolean', 'Date', 'Location', 'Object']);
```

Comment:
nit: I would extract this out of the function so you don't have to instantiate it even function call